build/Targets/VSL.Versions.targets

```diff
@@ -82,6 +82,16 @@
     </Otherwise>
   </Choose>
 
+  <PropertyGroup>
+    <GitHeadSha>Not found</GitHeadSha>
+    <DotGitDir>$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)../../.git'))</DotGitDir>
+    <HeadFileContent Condition="Exists('$(DotGitDir)')">$([System.IO.File]::ReadAllText('$(DotGitDir)/HEAD').Trim())</HeadFileContent>
+    <RefPath Condition="$(HeadFileContent.StartsWith('ref: '))">$(DotGitDir)/$(HeadFileContent.Substring(5))</RefPath>
```

Comment:
I believe packed refs are a setup the user has to opt into correct?  It won't for instance just sneak up on us one day when git hits a specific limit of files. Still someting we have to deal with but want to make I understand the problem here.

The other item that Andy suggested was to just use the git command line and print out the head.  I had some push back on that because I didn't want to presume git was available on the command line.  Feels like an easy way to break a downstream build component.